jupyterkernelgen/jupyterkernelgen.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+import os
+import glob
+import shutil
+import sys
+import readline
+import subprocess
+
+# allow tab completion in input()
+readline.set_completer_delims(' \t\n=')
+readline.parse_and_bind("tab: complete")
+
+# define ANSI escape codes for coloured text
+class colors:
+    HEADER = '\033[95m'
+    OK = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+
+def check_for_conda():
```

Comment:
Can you provide comment blocks for each function 
    """
    Nice easy description of what this function does. 

    :param FIRST: what is this thing?
    :param SECOND: Why do you need this?
    :return: TYPE. Description of what you are returning
    """

Please add type hinting for the signature.

Reply:
Done for all functions